app/controllers/orders_controller.rb

```diff
@@ -1,38 +1,41 @@
 class OrdersController < ApplicationController
+  include PurchasesHelper
+
+  before_action :validate_product, only: :new
+
   def new
-    @order = Order.new(product: Product.find(params[:product_id]))
+    @order = Order.new(product: product)
   end
 
   def create
     child = Child.find_or_create_by(child_params)
     @order = Order.create(order_params.merge(child: child, user_facing_id: SecureRandom.uuid[0..7]))
-    if @order.valid?
+    if child.valid? && @order.valid?
```

Comment:
`@order` won't be valid if `@child` isn't valid, but want to run validation on the `@child` anyway in order to populate `child.errors`